### PR TITLE
Add actual integration queries to table

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
Adds the ability to dynamically load integrations into the installed integrations table, and reactivates the tab. Also adds some terminal spacing to the S3 setup page since it was getting clipped by the bottom bar.

TODO: Cypress tests

Empty Table:
image

When an empty test integration is added that matches the data source:
image

### Issues Resolved
N/A

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Swiddis** [comment]
Depends on #1530

File: server/adaptors/integrations/types.ts [hunk 1]
```diff
@@ -124,6 +124,7 @@ interface AssetReference {
   assetId: string;
   isDefaultAsset: boolean;
   description: string;
+  status?: string;
```
Comment (ps48) [inline server/adaptors/integrations/types.ts:127]: Can the status be from an enum?

File: public/components/integrations/components/setup_integration.tsx [hunk 2]
```diff
@@ -415,6 +415,9 @@ export function SetupIntegrationForm({
               </EuiFormRow>
             </>
           ) : null}
+          {/* Bottom bar will overlap content if there isn't some space at the end */}
+          <EuiSpacer />
```
Comment (ps48) [inline public/components/integrations/components/setup_integration.tsx:419]: nit. Euispacer has props for size="l"/"xl". May be use that.

File: public/components/integrations/components/added_integration.tsx [hunk 3]
```diff
@@ -27,11 +27,24 @@ import React, { useEffect, useState } from 'react';
...
+export const IntegrationHealthBadge = ({ status }: { status?: string }) => {
+  switch (status) {
+    case undefined:
+      return <EuiHealth color="warning">Unknown</EuiHealth>;
+    case 'available':
+      return <EuiHealth color="success">Active</EuiHealth>;
+    case 'partially-available':
+      return <EuiHealth color="warning">Partially Available</EuiHealth>;
+    default:
+      return <EuiHealth color="danger">Critical</EuiHealth>;
```
Comment (ps48) [inline public/components/integrations/components/added_integration.tsx:44]: not a blocker: Is there any documentation or comment where i can learn about how we determine the status? 
As a user I would like to learn more about what statuses represent

**Review by ps48 — APPROVED** [review]

**Review by paulstn — APPROVED** [review]

File: public/components/integrations/components/setup_integration.tsx (hunk 2 above)
Comment (Swiddis) [reply public/components/integrations/components/setup_integration.tsx:419]: Tried xxl but it was still too small, 2 default spacers did the trick

File: public/components/integrations/components/added_integration.tsx (hunk 3 above)
Comment (Swiddis) [reply public/components/integrations/components/added_integration.tsx:44]: We documented it in the design somewhere but I don't think we've added it to OSS docs, I'll see about finding a home for the info

File: server/adaptors/integrations/types.ts (hunk 1 above)
Comment (Swiddis) [reply server/adaptors/integrations/types.ts:127]: I have a few things to fix with status so I made #1545, will work on it soon

**Pull request merged**